File: language/benchmarks/tokens/algorithms/linear-search/linear-search.ts
function linearSearch(target: number, xs: number[]): number {
  return linearSearchFrom(0, target, xs);
}

function linearSearchFrom(idx: number, target: number, xs: number[]): number {
  if (xs.length === 0) return -1;
  const [head, ...tail] = xs;
  if (head === target) return idx;
  return linearSearchFrom(idx + 1, target, tail);
}

File: language/benchmarks/tokens/algorithms/linear-search/linear-search.py
def linear_search(target: int, xs: list[int]) -> int:
    return linear_search_from(0, target, xs)


def linear_search_from(idx: int, target: int, xs: list[int]) -> int:
    if len(xs) == 0:
        return -1
    head, *tail = xs
    if head == target:
        return idx
    return linear_search_from(idx + 1, target, tail)
